File: src/main/java/io/github/wntopia/gikipedia/server/domain/auth/service/SigninService.java
package io.github.wntopia.gikipedia.server.domain.auth.service;

import io.github.wntopia.gikipedia.server.domain.auth.dto.request.DataGsmOAuthCallbackReqDto;
import jakarta.servlet.http.HttpSession;
import org.springframework.http.ResponseEntity;

public interface SigninService {

  ResponseEntity<Void> execute(DataGsmOAuthCallbackReqDto reqDto, HttpSession session);
}

File: src/main/kotlin/io/github/wntopia/gikipedia/server/domain/auth/service/SigninService.kt
package io.github.wntopia.gikipedia.server.domain.auth.service

import io.github.wntopia.gikipedia.server.domain.auth.dto.request.DataGsmOAuthCallbackReqDto
import jakarta.servlet.http.HttpSession
import org.springframework.http.ResponseEntity

interface SigninService {
    fun execute(
        reqDto: DataGsmOAuthCallbackReqDto,
        session: HttpSession,
    ): ResponseEntity<Void>
}
